OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/CourseModalDialogFragment.kt

```diff
@@ -77,32 +106,114 @@ class CourseModalDialogFragment : RoboDialogFragment() {
             billingProcessor =
                 BillingProcessor(requireContext(), object : BillingProcessor.BillingFlowListeners {
                     override fun onPurchaseCancel() {
-                        enableUpgradeButton(true)
+                        iapViewModel.endLoading()
```

Comment:
IMO, we can remove this as loading will be handled by observers and viewModel.
Please update others too.

Reply:
updated